src/libexec/zmlogprocess

```diff
@@ -246,7 +246,7 @@ sub run() {
                 }
             }
         } elsif ($app =~ /^postfix/o) {
-            if ($msg =~ /^([A-F0-9]+):/) {
+            if ($msg =~ /^([A-F0-9]{6,}|[0-9B-DF-HJ-NP-TV-Zb-df-hj-np-tv-z]{10}z[0-9B-DF-HJ-NP-TV-Zb-df-hj-np-tv-y]+):/) {
```

Comment:
That would match lines where `$msg` contains, for example,
```
NOQUEUE: reject: RCPT from unknown[117...
````
and `$qid` would be set to `NOQUEUE` and the processing would go forward.
In general, since the queue id has a specific format which quite likely will not change, why choose something more generic which, especially with something like `[a-z\d]+`, could match basically any alphanumeric sequence. Previously `[A-F0-9]` narrowed the range of possible matches significantly, but using the full range of characters of the long format would allow for a multitude of wrong possibilities.
Now, the way I see it, since (a) the logs are generated by Postfix and the QID regex will always match a QID generated by Postfix if it adheres to the specs, (b) I believe to be extremely unlikely that a format might be modified by Postfix or a third one might be introduced in the future and (c) this portion of code has been barely touched over all these years, so using a simpler regex to the benefit of someone less familiar with the syntax of a more precise one is hardly necessary, what might be the benefits of simplifying the regex losing accuracy?
(A link to the Postfix definition of the format might be included with a comment.)
A point might be performance, but this commands examines logs over a 10-minute window at every run, so I think the benefits might be negligible if there are any. I actually checked on a 100000 line sample containing just lines where the match would be performed, and a more relaxed test pattern ( `/^([^N][^:]+):/` ) actually took more time (I think because the more precise pattern failed more quickly where it wasn't supposed to match, but frankly the results didn't differ that much).
I am not trying to be overly punctilious, but in general I prefer to lose accuracy only when there's a good reason to do so, such as performance, input from human sources which might come up with unpredictable variations, etc.
Do you see something I am not seeing?

Reply:
(BTW I've noticed an imprecision in the lenght of the first portion in the regex from one of your comments below. I'll fix that soon.)

Reply:
Oops, matching NOQUEUE would be undesirable.  I'm not fully opposed to being strict(er), but historically log parsers benefit from being more flexible than not.  Regardless, if we're going to be more strict, I'd at least like us to be consistent about how we're matching and I'd currently prefer something like what was proposed for other parts of this change:
```
my $RECHR = qr/[\dBCDFGHJKLMNPQRSTVWXYZ]/i;                     
my $REGEX_POSTFIX_QID = qr{(?:[A-F0-9]{6,}|$RECHR{10,}?z$RECHR+)};
```
I simply find this more readable when looking quickly at the code.